clippy_lints/src/if_let_redundant_pattern_matching.rs

```diff
@@ -0,0 +1,81 @@
+use rustc::lint::*;
+use rustc::hir::*;
+use utils::{paths, span_lint_and_then, match_path, snippet};
+
+/// **What it does:*** Lint for redundant pattern matching over `Result` or `Option`
+///
+/// **Why is this bad?** It's more concise and clear to just use the proper utility function
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rustc
```

Comment:
Please add empty lines before and after the code blocks for this to be correctly rendered as markdown